Undefined symbol _ZN3fmt2v86detail13error_handler8on_errorEPKc

Upgrading fmt from 8.0.1 to 8.1.0 breaks a CI run here: https://github.com/flatsurf/flatsurf/runs/4698212878?check_suite_focus=true#step:6:554

In that CI run, we are trying to load a library, libintervalxt, which was [built against fmt 8.0.1](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=380913&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=688) with [fmt 8.1.0 installed](https://github.com/flatsurf/flatsurf/runs/4698212878?check_suite_focus=true#step:6:432).

While the missing symbol is not part of the public API of fmt, it used to have `FMT_API` set in 8.0.1 but not anymore in 8.1.0. At least in the setup that the conda-forge distribution uses, this made that symbol disappear from the library. Since libraries such as libintervalxt build against the old ABI (without using this symbol explicitly in the code of course) this seems like a breaking ABI change was introduced in the 8.1.0 release.

Or maybe conda-forge's libintervalxt and fmt are using the wrong compiler flags somehow?

Same issue for Debian, https://release.debian.org/britney/pseudo-excuses-experimental.html

```
fmtlib (8.0.1+ds1-4 to 8.1.0+ds1-1)
Migration status for fmtlib (8.0.1+ds1-4 to 8.1.0+ds1-1): BLOCKED: Rejected/violates migration policy/introduces a regression
Issues preventing migration:
autopkgtest for bear/3.0.17-1: amd64: Regression ♻ (reference ♻), arm64: Regression ♻ (reference ♻)
autopkgtest for ceph/16.2.7+ds-1: amd64: Regression ♻ (reference ♻), arm64: Regression ♻ (reference ♻)
autopkgtest for cryfs/0.11.0-2: amd64: Regression ♻ (reference ♻), arm64: Regression ♻ (reference ♻)
autopkgtest for spdlog/1:1.9.2+ds-0.2: amd64: Regression ♻ (reference ♻), arm64: Regression ♻ (reference ♻)
autopkgtest for therion/6.0.4-1: amd64: Regression ♻ (reference ♻), arm64: Regression ♻ (reference ♻)
Additional info:
Cannot be tested by piuparts (not a blocker) - (no link yet)
1 days old
```

spdlog seems to use this symbol as well.

```
/usr/bin/c++ -O2 CMakeFiles/spdlog-utests.dir/test_file_helper.cpp.o CMakeFiles/spdlog-utests.dir/test_file_logging.cpp.o CMakeFiles/spdlog-utests.dir/test_daily_logger.cpp.o CMakeFiles/spdlog-utests.dir/test_misc.cpp.o CMakeFiles/spdlog-utests.dir/test_eventlog.cpp.o CMakeFiles/spdlog-utests.dir/test_pattern_formatter.cpp.o CMakeFiles/spdlog-utests.dir/test_async.cpp.o CMakeFiles/spdlog-utests.dir/test_registry.cpp.o CMakeFiles/spdlog-utests.dir/test_macros.cpp.o CMakeFiles/spdlog-utests.dir/utils.cpp.o CMakeFiles/spdlog-utests.dir/main.cpp.o CMakeFiles/spdlog-utests.dir/test_mpmc_q.cpp.o CMakeFiles/spdlog-utests.dir/test_dup_filter.cpp.o CMakeFiles/spdlog-utests.dir/test_fmt_helper.cpp.o CMakeFiles/spdlog-utests.dir/test_stdout_api.cpp.o CMakeFiles/spdlog-utests.dir/test_backtrace.cpp.o CMakeFiles/spdlog-utests.dir/test_create_dir.cpp.o CMakeFiles/spdlog-utests.dir/test_cfg.cpp.o CMakeFiles/spdlog-utests.dir/test_time_point.cpp.o CMakeFiles/spdlog-utests.dir/test_stopwatch.cpp.o CMakeFiles/spdlog-utests.dir/test_errors.cpp.o -o spdlog-utests  /usr/lib/x86_64-linux-gnu/libspdlog.so.1.9.2 -lpthread /usr/lib/x86_64-linux-gnu/libfmt.so.8.1.0 -Wl,--as-needed 
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libspdlog.so.1.9.2: undefined reference to `fmt::v8::detail::error_handler::on_error(char const*)'
```

--- a/include/fmt/core.h
+++ b/include/fmt/core.h
@@ -601,7 +601,8 @@ struct error_handler {
   constexpr error_handler(const error_handler&) = default;
 
   // This function is intentionally not constexpr to give a compile-time error.
-  void on_error(const char* message) { throw_format_error(message); }
+  // This function is marked as FMT_API for backwards compatibility, see #2695.
+  FMT_NORETURN FMT_API void on_error(const char* message) { throw_format_error(message); }
 };
 FMT_END_DETAIL_NAMESPACE
 

